includes/admin.form.inc

```diff
@@ -41,7 +41,12 @@ function islandora_multi_importer_admin_form($form, $form_state) {
     '#description' => t('This option also requires corresponding Drupal permissions to be correctly set.'),
     '#default_value' => variable_get('islandora_multi_importer_extrads', FALSE),
   );
-
+  $form['islandora_multi_importer_generalsettings_fieldset']['islandora_multi_importer_temp_dir'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Temporary files storage directory'),
+    '#description' => t('Path to where to store temporary files under Drupal\'s <em>private://</em> directory. This is used for storing batch ingest files, generated metadata files, and other temporary files. Do not include leading or trailing slash. The <em>private://</em> directory location is configured at <a href="/admin/config/media/file-system">Configuration > Media > File system</a>' ),
+    '#default_value' => variable_get('islandora_multi_importer_temp_dir', 'islandora-multi-importer'),
+  );
```

Comment:
@patdunlavey so location is now fixed to private? What happens if people, that are already using this, have not access to private, or creating this? Should we have the option to decide also the location (private://, temp:// or public://?)

Reply:
I thought about that. The location of private:// is itself configurable, and by default is located under /sites/default/files/private, with protection from a .htaccess file. Everybody should have access to that. I can't think of a reason why someone would want these files to be public. Access to your data should be mediated by xacml policy, etc. We should not create a security leak.
That's my thinking.

Reply:
Of course islandora_batch itself insists in putting the MODS xml files into public://. I didn't figure out how to override that.

Reply:
Could we verify if private exits and it can be written? Also, maybe time to add some documentation. Is that something we could add in this pull?

Reply:
Well public is not "public", in the terms of anonymous user facing public, but it does add some risks, i agree with you

Reply:
I see that the private:// scheme may not be defined, and in that case there is not a graceful failover to public://. So I'm going to have to take this back to the woodshed!

Reply:
Cool, thanks!

Reply:
Woodshedding done. It now checks to see if the "private" stream wrapper is defined. If not, it will not save the temporary file directory and throw a validation error when you try. I decided not to failover to "public".

Reply:
@patdunlavey sorry for not merging this before. I'm totally ok but got a request/usecase to allow temps to actually love in temporary://, so fixing this to private is giving me some issues. I really need to allow people to select which streamwrapper they prefer, not only the path. Give me this weekend to add a pull with that change. Sorry man.

Reply:
@DiegoPino I added the ability to select the stream wrapper. I've tested the configuration functionality, but have not actually tested running an import!